modules/hcp-eks-client/templates/consul.tpl

```diff
@@ -28,17 +37,17 @@ externalServers:
 server:
   enabled: false
 
-client:
-  enabled: true
-  join: ${consul_hosts}
-  nodeMeta:
-    terraform-module: "hcp-eks-client"
-
 connectInject:
   transparentProxy:
-    defaultEnabled: true
+    defaultEnabled: false
   enabled: true
   default: true
+%{ if !consul_client_agent ~}
+  consulNode:
+    meta:
+      terraform-module: "hcp-eks-client"
+%{ endif ~}
+
 
 controller:
```

Comment:
note-to-self: remove this for agentless (connectInject.enabled = true should suffice).